src/ledger/parse/utils.js

```diff
@@ -58,13 +58,32 @@ function isPartialPayment(tx) {
 function parseDeliveredAmount(tx: Object): Amount | void {
   let deliveredAmount;
 
-  // TODO: Workaround for existing rippled bug where delivered_amount may not be
-  // provided for account_tx
-  if (tx.TransactionType === 'Payment') {
-    if (tx.meta.delivered_amount) {
+  if (tx.TransactionType === 'Payment' &&
+      tx.meta.TransactionResult === 'tesSUCCESS') {
+
+    if (tx.meta.delivered_amount &&
+        tx.meta.delivered_amount === 'unavailable') {
+      deliveredAmount = 'unavailable';
```

Comment:
You could also break the else-if chain into just a sequence of if-returns which would make the comments less awkward